spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -1500,12 +1504,12 @@ public void run() {
 			try {
 				if (System.currentTimeMillis() - JmsOutboundGateway.this.lastSend >
 						JmsOutboundGateway.this.idleReplyContainerTimeout
-						&& JmsOutboundGateway.this.replies.size() == 0 &&
-						JmsOutboundGateway.this.replyContainer.isRunning()) {
+						&& JmsOutboundGateway.this.replies.isEmpty() &&
+						Objects.requireNonNull(JmsOutboundGateway.this.replyContainer).isRunning()) {
 
 					logger.debug(() -> getComponentName() + ": Stopping idle reply container.");
 					JmsOutboundGateway.this.replyContainer.stop();
-					JmsOutboundGateway.this.idleTask.cancel(false);
+					Objects.requireNonNull(JmsOutboundGateway.this.idleTask).cancel(false);
```

Comment:
This is not OK.
We have to extract `JmsOutboundGateway.this.idleTask` into a local variable.
Guard with `if (idleTaskToCancel != null)` and so on the rest against this variable.
But yeah... `JmsOutboundGateway.this.idleTask = null;` is still correct.